src/devices_models/device_constructors/thermalgeneration_constructor.jl

```diff
@@ -1,3 +1,16 @@
+# TODO call this for non-thermal generators too?
```

Comment:
we can but that is probably a later thing to do. Other devices might have other implementations of what attributed to look for.